enhancements/sig-architecture/64-placementStrategy/README.md

```diff
@@ -0,0 +1,220 @@
+
+### Goals
+
+1. Add a new field placementStrategy to the placement APIs to let end uer define the roll out strategy for the selected clusters.
+
+### Non-Goals
+
+1. The new Placement APIs field placementStrategy has no control over applying the workloads.
+1. The new Placement APIs field placementStrategy has no control over scheduling the workloads.
+
+## Proposal
+
+### User Stories
+
+#### Story 1: As end user I want to define a subset of the selected clusters as a canary clusters.
+#### Story 2: As end user I want to divide the selected clusters to groups then a workloads applier APIs can use those cluster groups to roll out the workloads. 
+
+### Placement placementStrategy API
+
+Adding PlacementStrategy to PlacementSpec. PlacementStrategy has a labelSelector to identify canary clusters. CanaryClusters must be a subset of the selected clusters based on predicates and ClusterSets. MaxConcurrency is a percentage of the total selected clusters that will divide the clusters to groups(batches). Example; for a total 100 clusters selected, MaxConcurrency equal to 20% will divide the selected clusters into 5 groups each group consist of 20 clusters.
+```go
+type PlacementSpec struct {
+	// ClusterSets represent the ManagedClusterSets from which the ManagedClusters are selected.
+	// +optional
+	ClusterSets []string `json:"clusterSets,omitempty"`
+
+	// Predicates represent a slice of predicates to select ManagedClusters. The predicates are ORed.
+	// +optional
+	Predicates []ClusterPredicate `json:"predicates,omitempty"`
+            ...
+ 
+	// PlacementStrategy define canary clusters as subset of the selected clusters and divide the selected clusters to groups.
+	// +optional
+	PlacementStrategy PlacementStrategy `json:"placementStrategy,omitempty"`
+}
+
+type PlacementStrategy struct {
+	// Label selector for canary clusters as subset of the selected clusters from predicates and ClusterSets.
+	// +optional
+	CanaryClusters metav1.LabelSelector `json:"canaryClusters,omitempty"`
+
+	// MaxConcurrency is a percentage of the total selected clusters that will divide the clusters to groups.
+	// ex; for a total 100 clusters selected, MaxConcurrency equal to 20% will divide the selected clusters to
+	//     5 groups each group should have 20 clusters.
+	// +kubebuilder:maxConcurrency:Required
+	// +kubebuilder:maxConcurrency:Enum=5%;10%;15%;20%;25%;50%;100%
+	// +kubebuilder:default:=100%
+	// +optional
+	MaxConcurrency string `json:"maxConcurrency,omitempty"`
+}
+```
+
+Adding NumberOfClusterGroups to the PlacementStatus API 
+```go
+type PlacementStatus struct {
+	// NumberOfClusterGroups determined by placement strategy. If there is no placement strategy defined all selected clusters will be considered in 1 group
+  // +kubebuilder:default:=1
+	// +optional
+	NumberOfClusterGroups int32 `json:"numberOfClusterGroups"`
```

Comment:
How about we create 3 placements, each as a group. And we can set predicate and percentage on each placement. So each placement can select a certain percentage of clusters with the same predicate. We then could set an anti-affinity across placement to ensure each selection of placement is exclusive.

Reply:
Just to clarify, a better way of saying what I meant is "no shuffling between groups due to cluster changes (labels, new cluster, deleted cluster etc)". When the placement spec changes (predicates, strategy), we probably don't want to promise any kind of stability. Users have to expect all groups to be reformed.

Reply:
@qiujian16 By "as a group", do you mean the placements can share the same predicates or we have to duplicate them? My understanding of the intent of this enhancement is to capture the essence of the grouping concepts and knobs in one place rather than spread across many CRs. I suppose variable size grouping could be useful in some cases. If it's part of the goal, we can have a percentage array in the strategy to realize that.

Reply:
I think we can have a field in the placement defining that this placement is affinity to another placement, so you do not need to duplicate the predicate

Reply:
From a user perspective it might seem disjoint, if a placement creates and references other placements. It's understood that a placement creates multiple decisions and extending that to include groupings seems like a logical step.